backend/server/game_api.js

```diff
@@ -0,0 +1,637 @@
+    };
+});
+
+
+// delete a game
+gameRouter.delete('/games/:game_id', async (req, res) => {
+    try {
+        const game = await pool.query('select * from games where id = $1', [req.params.game_id]);
+        const gameDetails = game.rows[0];
+
+        if (req.user.id !== gameDetails.user2) {
+            return res.status(401).json({msg: 'You are not the correct opponent player'});
+        }
+
+        if (gameDetails.state !== 'invited') {
+            return res.status(401).json({msg: 'Cannot delete an active game'});
+        }
+
+        await pool.query('delete from games where id = $1;', [req.params.game_id]);
+        return res.status(200).json({msg: 'game deleted by opponent'});
+
+    } catch(e) {
+        res.status(500).json({msg: 'Server error'});
+    };
+});
+
+
+// Placing ships
+function checkShipPlacement(start_row, end_row, start_col, end_col, shipInDb) {
+    if (shipInDb.start_row === shipInDb.end_row) {
+        if ((start_row === shipInDb.start_row && start_col === (shipInDb.start_col - 1)) || (start_row === shipInDb.start_row && start_col === (shipInDb.end_col + 1))) {
+            return 1;
+        }
+
+        if (start_row === (shipInDb.start_row + 1) && (start_col >= (shipInDb.start_col - 1) && start_col <= (shipInDb.end_col + 1))){
+            return 1;
+        }
+
+        if (start_row === (shipInDb.start_row - 1) && (start_col >= (shipInDb.start_col - 1) && start_col <= (shipInDb.end_col + 1))){
+            return 1;
+        }
+
+        if (start_row < shipInDb.start_row && end_row > shipInDb.start_row && (start_col >= (shipInDb.start_col - 1) && start_col <= (shipInDb.end_col + 1))) {
+            return 1;
+        }
+    } else if (shipInDb.start_row !== shipInDb.end_row) {
+        if (start_col === (shipInDb.start_col - 1) && (start_row === shipInDb.start_row || start_row === shipInDb.end_row)) {
+            return 1;
+        }
+
+        if (start_col === (shipInDb.start_col + 1) && (start_row === shipInDb.start_row || start_row === shipInDb.end_row)) {
+            return 1;
+        }
+
+        if (start_row === (shipInDb.start_row - 1) && (start_col >= (shipInDb.start_col - 1) && start_col <= (shipInDb.start_col + 1))) {
+            return 1;
+        }
+
+        if (start_row === (shipInDb.end_row + 1) && (start_col >= (shipInDb.start_col - 1) && start_col <= (shipInDb.start_col + 1))) {
+            return 1;
+        }
+
+        if (start_col < shipInDb.start_col && end_col > shipInDb.start_col && (start_row >= (shipInDb.start_row - 1) && start_row <= (shipInDb.end_row + 1))) {
+            return 1;
+        }
+    }
+    return 0;
+};
+
+
+async function checkShips(req, dimension, ship_size) {
+    const userShips = await pool.query('select size from ships where game_id = $1 and user_id = $2', [req.params.game_id, req.user.id]);
+    const gameShips = shipsAmount[dimension];
+    const placedShips = userShips.rows.map(ship => ship.size);
+
+    let count = 0;
```

Comment:
instead of this `for` loop, you can use `placedShips.filter(....).length`. 
Also, please give a meaningful name to this function such as `verifyReminingShipsOfSize`.